app/views/unattended/provisioning_templates/registration/global_registration.erb

```diff
@@ -3,8 +3,46 @@ kind: registration
 name: Global Registration
 model: ProvisioningTemplate
 -%>
+#!/bin/sh
+<%
+    curl_headers = ["-H 'Content-Type: application/json'", "-H 'Accept: application/json'", "-H 'Authorization: Bearer #{@auth_token}'"]
+    curl_vars = { headers: curl_headers, organization: @organization, location: @location, hostgroup: @hostgroup }
+%>
+
+# Template parameters:
+<%= "#  - User: [#{@user.login}]" if @user %>
+<%= "#  - Organization: [#{@organization.name}]" if @organization %>
+<%= "#  - Location: [#{@location.name}]" if @location %>
+<%= "#  - Hostgroup: [#{@hostgroup.name}]" if @hostgroup %>
 
 if ! [ $(id -u) = 0 ]; then
     echo "Please run as root"
     exit 1
 fi
+
+if [ -f /etc/os-release ] ; then
+  . /etc/os-release
+fi
+
+echo "#"
+echo "# Running registration"
+echo "# Hostname: $HOSTNAME"
```

Comment:
Here `$HOSTNAME` may be different from the `$HOSTNAME` in the curl script which makes debugging harder. `$HOSTNAME` is always available, but is the system hostname. It may be a short name without a domain, but `hostname -f` also performs a resolver lookup (typically via DNS).

I highly recommend to read the `THE FQDN` section in `man hostname(1)` to understand its limitations.